orchestrator-bundle/orc8r-certifier-operator/src/charm.py

```diff
@@ -312,27 +368,148 @@ def _on_install(self, event: InstallEvent) -> None:
+                event.defer()
+                return
+            if not self._root_private_key_is_stored:
+                self.unit.status = WaitingStatus("Waiting for root private key to be generated")
+                event.defer()
+                return
+            if not self._application_private_keys_are_stored:
+                self.unit.status = WaitingStatus(
+                    "Waiting for application private key to be generated"
+                )
+                event.defer()
+                return
+            if not self._root_csr_is_stored:
+                self._generate_root_csr()
+                if self._certificates_relation_created:
+                    self._request_certificate_based_on_stored_csr()
+                    self.unit.status = WaitingStatus(
+                        "Waiting to receive new certificate from provider"
+                    )
+                else:
+                    self.unit.status = BlockedStatus(
+                        "Waiting for tls-certificates relation to be created"
+                    )
+            if not self._stored_root_csr_matches_config:
+                old_csr = self._root_csr
+                self._generate_root_csr()
+                if self._certificates_relation_created:
+                    self.tls_certificates_requirer.request_certificate_renewal(
+                        old_certificate_signing_request=old_csr.encode(),  # type: ignore[union-attr]  # noqa: E501
+                        new_certificate_signing_request=self._root_csr.encode(),  # type: ignore[union-attr]  # noqa: E501
+                    )
+                    self.unit.status = WaitingStatus(
+                        "Waiting to receive new certificate from provider"
+                    )
+                else:
+                    self.unit.status = BlockedStatus(
+                        "Waiting for tls-certificates relation to be created"
+                    )
+            if not self._application_certificates_are_stored:
+                self._generate_application_certificates()
+                self._push_application_certificates()
+            if not self._stored_application_certificate_matches_config:
+                self._generate_application_certificates()
+                self._push_application_certificates()
+        else:
+            if not self._container.can_connect():
+                self.unit.status = WaitingStatus("Waiting for container to be ready")
+                event.defer()
+                return
+            if not self._root_csr_is_stored:
+                self.unit.status = WaitingStatus("Waiting for leader to generate root csr")
+                event.defer()
+                return
+            if not self._stored_root_csr_matches_config:
+                self.unit.status = WaitingStatus("Waiting for leader to generate new root csr")
+                event.defer()
+                return
```

Comment:
like above this also can be one condition, the root csr is new in both cases, isn't it? or do we need to distinguish between the two?
```suggestion
            if not self._root_csr_is_stored or not self._stored_root_csr_matches_config:
                self.unit.status = WaitingStatus("Waiting for leader to generate a new root csr")
                event.defer()
                return
```

Reply:
`_stored_application_certificate_matches_config` raises if there is no stored certificate, hence why I placed them on 2 separate lines

Reply:
Fixed

Reply:
> `_stored_application_certificate_matches_config` raises if there is no stored certificate, hence why I placed them on 2 separate lines

Was this reply intended to this comment? if so I don't fully get it :D

Reply:
Yeah it's fixed now anywaybut yes what I meant was that the second `if` would raise an exception if the root csr isn't stored but like Ghislain mentioned above, we will never get in that situation given the first if would catch it.